src/lib/scheduler/node_queue_scheduler.cpp

```diff
@@ -68,6 +68,18 @@ void NodeQueueScheduler::wait_for_all_tasks() {
 
     std::this_thread::sleep_for(std::chrono::milliseconds(10));
   }
+
+  auto queue_check_runs = size_t{0};
+  for (auto& queue : _queues) {
+    // The following assert checks that we are not looping forever. The empty() check can be inaccurate for concurrent
+    // queues when many tiny tasks have been scheduled (see MergeSort scheduler test). When this assert is triggered in
+    // other situations, there have probably been new tasks added after wait_for_all_tasks() was called.
+    Assert(queue_check_runs < 1'000, "Queues are not empty but all registered tasks have already been processed.");
+    while (!queue->empty()) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+    ++queue_check_runs;
```

Comment:
Do I get this right and you increment `queue_check_runs` once whenever a queue is not empty rather than once for each `empty()` call/sleep. How will this variable ever be large/larger than the number of queues?

Reply:
Argh, rewrote the loop from a `while (true)` loop. Thanks. Will rewrite the check.

Reply:
Should be addressed.